src/PermissionRegistrar.php

```diff
@@ -124,32 +121,50 @@ private function loadPermissions()
             return;
         }
 
-        $this->permissions = $this->cache->remember(self::$cacheKey, self::$cacheExpirationTime, function () {
-            // make the cache smaller using an array with only required fields
-            return $this->getPermissionClass()->select('id', 'id as i', 'name as n', 'guard_name as g')
-                ->with('roles:id,id as i,name as n,guard_name as g')->get()
-                ->map(function ($permission) {
-                    return $permission->only('i', 'n', 'g') +
-                        ['r' => $permission->roles->map->only('i', 'n', 'g')->all()];
+        $alias = ['id'=>'i', 'name'=>'n', 'guard_name'=>'g', 'label' => 'l', 'type' => 'y', 'description' => 'd', 'descrip' => 'e'];
```

Comment:
What do you think about possibly defining the `$alias` array in the package config as I suggested on the issue #1836?

Maybe even the columns of the roles as well (we would retrieve the whole model and cache only the attributes defined on the config).

We can use the values you put here as a default on the configuration, but that would allow for users to change them according to their needs, maybe having even smaller caches (e.g. if they don't need to cache a specific column, or defining an alias for custom columns as well).

That way this package doesn't need to worry about merging this values with the custom attributes of the models, having to worry about ignoring others (we could remove the `$except` array, if the attribute is not mapped on the config, we just don't cache it) and we just leave for the end user to edit their configuration according to their needs on each individual project.

We could set a `models` (or some other name, idk) to the `cache` tag on permissions:

```php
[
    'cache' => [
        'models' => [
            'role' => [
                'id' => 'i',
                'name' => 'n',
                'guard_name' => 'g',
            ],
            'permission' => [
                'id' => 'i',
                'name' => 'n',
                'guard_name' => 'g',
                'label' => 'l',
                'type' => 'y',
                'description' => 'd',
                'descrip' => 'e'
            ],
        ],
    ],
]
```

Reply:
Or even in the model class as a static property "$attributesAlias" or " $cacheAttributesAlias" by example

Reply:
At first I considered that, but almost everyone only uses the library with normal fields, adding that configuration could complicate things more(the aliases cannot be repeated, it makes unexpected errors), 
with this change make cache even smaller(thanks for the idea @danilopinotti), 
also the cache is not visible to the developer, if @drbyte thinks it would be a good idea, I can add it

Reply:
My idea was not necessarily about customizing the aliases that you've already setup, although it would allow for that as well, but to allow to define shorter tags for custom fields, from what I see, the PR fixes the problem of custom attributes by retrieving all the attributes of the model, ignoring the timestamps (that I personally don't need, but may be useful for someone, idk), but it uses the full column name on it, so say I have 3 columns that I need to add to the cache, it could increase its size drastically, but with the ability to set the aliases, I could use the same idea you used with a single or a couple of letters only.
I get your concern about making it complicated and causing unexpected errors, but at least the problem of repeated aliases could be fixed by just defining the array in a flipped way (with the alias as keys).
With a customized option, I could shrink my cache as well, since I don't really need the `label`, `type`, `description` and `descrip` columns, but need the `module_id`.
I may be wrong, but I guess users don't mess that much with default config variables unless they need to do so, so I don't think people would have problems with this approach (or the one suggested by @danilopinotti)

Reply:
I undestand your idea, but i need @drbyte point of view(he makes the merge)

>just defining the array in a flipped way (with the alias as keys)

That is good idea

> since I don't really need the label, type, description and descrip columns, but need the module_id

Don't worry, if you don't have any of these fields your cache will not change size, because if they don't exist they are skipped, `modulo_id` will be a small weight, initially my cache weighed 3.1mb now with this change it weighs 60kb and I have 3 custom fields, I achieved a 99% decrease

Reply:
> just defining the array in a flipped way (with the alias as keys)

I followed your idea, looks a little faster now, @isecchin @danilopinotti please try